google-cloud-firestore/src/test/java/com/google/cloud/firestore/MapperTest.java

```diff
@@ -1513,6 +1554,13 @@ public void serializeLongBean() {
     assertJson("{'value': 1234567890123}", serialize(bean));
   }
 
+  @Test
+  public void serializeBigDecimalBean() {
+    BigDecimalBean bean = new BigDecimalBean();
+    bean.value = BigDecimal.valueOf(1.1);
+    assertEquals(mapAnyType("value", "1.1"), serialize(bean));
+  }
+
```

Comment:
Can we add another test in here to verifies a roundtrip with a value larger than a double? This will ensure that we're not losing in information during serialization. (You'll also need to add an `equals` method to `BigDecimalBean` for this test to work)
```suggestion
  @Test
  public void bigDecimalRoundTrip() {
    BigDecimal doubleMaxPlusOne = BigDecimal.valueOf(Double.MAX_VALUE).add(BigDecimal.ONE);
    BigDecimalBean a = new BigDecimalBean();
    a.value = doubleMaxPlusOne;
    Map<String, Object> serialized = (Map<String, Object>) serialize(a);
    BigDecimalBean b = convertToCustomClass(serialized, BigDecimalBean.class);
    assertEquals(a, b);
  }

```